### ISSUE TITLE
ERROR Could not find model with name ...

### ISSUE THREAD

**Issue by diggy128** [WilldooIT/Pentaho-reports-for-OpenERP#152]
Hi all and thank you for a wonderful product.

I'm having an issue.
I'm trying to print a report with a parameter (that works fine when running from the Report Designer) and I'm getting the following error:

> 2014-10-02 11:57:09,412 19708 ERROR xxx openerp.service.web_services: Exception: <Fault 0: "Failed to invoke method execute in class com.willowit.reporting.PentahoRenderer: Could not find model with name 'hr.schedule.detail'">
> Traceback (most recent call last):
>   File "/usr/lib/python2.7/site-packages/openerp-7.0_20140609_231020-py2.7.egg/openerp/service/web_services.py", line 712, in go
>     (result, format) = obj.create(cr, uid, ids, datas, context)
>   File "/usr/lib/python2.7/site-packages/openerp-7.0_20140609_231020-py2.7.egg/openerp/addons/pentaho_reports/core.py", line 250, in create
>     rendered_report, output_type = report_instance.execute()
>   File "/usr/lib/python2.7/site-packages/openerp-7.0_20140609_231020-py2.7.egg/openerp/addons/pentaho_reports/core.py", line 193, in execute
>     return self.execute_report()
>   File "/usr/lib/python2.7/site-packages/openerp-7.0_20140609_231020-py2.7.egg/openerp/addons/pentaho_reports/core.py", line 228, in execute_report
>     rendered_report = proxy.report.execute(proxy_argument).data
>   File "/usr/lib64/python2.7/xmlrpclib.py", line 1234, in **call**
>     return self.__send(self.__name, args)
>   File "/usr/lib64/python2.7/xmlrpclib.py", line 1588, in __request
>     verbose=self.__verbose
>   File "/usr/lib64/python2.7/xmlrpclib.py", line 1274, in request
>     return self.single_request(host, handler, request_body, verbose)
>   File "/usr/lib64/python2.7/xmlrpclib.py", line 1307, in single_request
>     return self.parse_response(response)
>   File "/usr/lib64/python2.7/xmlrpclib.py", line 1483, in parse_response
>     return u.close()
>   File "/usr/lib64/python2.7/xmlrpclib.py", line 798, in close
>     raise Fault(**self._stack[0])
> Fault: <Fault 0: "Failed to invoke method execute in class com.willowit.reporting.PentahoRenderer: Could not find model with name 'hr.schedule.detail'">
> 2014-10-02 11:57:09,503 19708 ERROR xxx openerp.netsvc:
> (<class 'xmlrpclib.Fault'>, <Fault 0: "Failed to invoke method execute in class com.willowit.reporting.PentahoRenderer: Could not find model with name 'hr.schedule.detail'">, <traceback object at 0x7f1cc4097878>)
> Traceback (most recent call last):
>   File "/usr/lib/python2.7/site-packages/openerp-7.0_20140609_231020-py2.7.egg/openerp/netsvc.py", line 296, in dispatch_rpc
>     result = ExportService.getService(service_name).dispatch(method, params)
>   File "/usr/lib/python2.7/site-packages/openerp-7.0_20140609_231020-py2.7.egg/openerp/service/web_services.py", line 654, in dispatch
>     res = fn(db, uid, *params)
>   File "/usr/lib/python2.7/site-packages/openerp-7.0_20140609_231020-py2.7.egg/openerp/service/web_services.py", line 760, in exp_report_get
>     return self._check_report(report_id)
>   File "/usr/lib/python2.7/site-packages/openerp-7.0_20140609_231020-py2.7.egg/openerp/service/web_services.py", line 738, in _check_report
>     netsvc.abort_response(exc, exc.message, 'warning', exc.traceback)
>   File "/usr/lib/python2.7/site-packages/openerp-7.0_20140609_231020-py2.7.egg/openerp/netsvc.py", line 71, in abort_response
>     raise openerp.osv.osv.except_osv(description, details)

The report is based on hr.schedule module by Michael Telahun Makonnen that provides the hr.schedule.detail model and as I said above, running the report from report designer 5.1.0 runs fine (with the parameter too).

Any pointers?

PS. I've seen this error with another module I'm developing alongside, but haven't got to it yet.

**Comment by richard-willdooit** [comment]
I suspect there will be more information in the tomcat logs (catalina.out)  Could you have a look for anything in there?  Do you get an error on the client screen?  Is it version 7 or 8 of odoo?  Is it object or sql?

Actually, looking deeper here are some thoughts:
- **ERROR xxx** in your trace _could_ be an old placeholder, make sure you are using the latest codebase.
- **xxx** IS a variable used by odoo in **hr_timesheet_invoice/report/account_analytic_profit.py** so this could be nothing to do with Pentaho.
- Check your service names on the actions.  The only reference to pentaho directly in the stack trace is at the service call - BUT it does NOT look like it has called Pentaho.  (catalina.out will help identify whether it is in fact getting to there).  It looks like it has called the standard report service.

This last point seems like the most likely.  Please check you have a unique service name for the report.

**Comment by diggy128** [comment]
Sorry, I should have mentioned that I've replaced my db name with xxx.
It is version 7 and accessing data via object.

Also, I just found out that there is an updated addon and java component for Pentaho 5.1 and after updating both the report runs just fine.
It was not clear that a minor version change could break things and I'm not too keen updating every minor version.

Sorry for the trouble and thank you for your time.
